### DISCUSSION TITLE
Miscellaneous rules

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#54] [Q&A]
While going through the tiddlyclip [issues](https://github.com/buggyj/tiddlyclip/issues/68#issuecomment-907766500), I found this list shared by you 

<img width="559" alt="image" src="https://user-images.githubusercontent.com/67494083/149888599-3b3b639d-e81e-43d8-9ead-775d48dddb74.png">

I am interested in knowing about these rules

1. extPdf,extMusic
2. youtubeRule
3. context
4. WebTextClipList
5. WebSnapRule, WebSnapClipList
6. WebLstRule (this I found in the tiddlyclip demo site)

If these rules are different from the already discussed rules, could you please share those rule tids and its functioning

**Comment by arunnbabu81** [comment]
Could you please tell me how to use the youtubeRule

**Comment by buggyj** [comment]
I use youtubeRule to create tracks for my media player - 
https://github.com/buggyj/tw5-xperimental/tree/master/xplugins/bj/mplayers 

I have not tried this with tiddlywiki greater than 5.1.17

**Reply by arunnbabu81** [reply]
may be that's beyond my skillset.

My usecase is this. I use [youtube annotation plug in by tobibeer](http://youtube.tiddlyspot.com/#Introduction%20To%20TiddlyWiki). In that i need to feed the youtube video id manually to a field called "yt-id" for that tiddler to become a youtube annotation tiddler.

For example, in this url "https://www.youtube.com/watch?v=KtCUr83XgyE" , id is "KtCUr83XgyE

 Can i extract the id of youtube video using tiddlyclip and add it to the "yt-id" field of the tiddler created. Can this be done using the method mentioned in this [discussion](https://github.com/buggyj/tiddlyclip-plugin/discussions/47#discussioncomment-1961270)

**Reply by arunnbabu81** [reply]
> I use youtubeRule to create tracks for my media player - https://github.com/buggyj/tw5-xperimental/tree/master/xplugins/bj/mplayers
> 
Can you please guide me how to you use this. Just want to try it out?

**Reply by buggyj** [reply]
It forms part of a media player 'edition' it would take a lot of explaining. I have playlists based on tags - each track can also have tiddlers displayed with it, as it is played. The example below show some Spanish music I am playing along with the songs lyrics and their English translation that I clipped from the web

![image](https://user-images.githubusercontent.com/2855795/151239473-9f790d24-b7d1-4e17-8e35-30fb96a53217.png)

**Reply by arunnbabu81** [reply]
> It forms part of a media player 'edition' it would take a lot of explaining. I have playlists based on tags - each track can also have tiddlers displayed with it, as it is played. The example below show some Spanish music I am playing along with the songs lyrics and their English translation that I clipped from the web
> 
> ![image](https://user-images.githubusercontent.com/2855795/151239473-9f790d24-b7d1-4e17-8e35-30fb96a53217.png)

Do you mind sharing a sample wiki containing the media player.

**Reply by buggyj** [reply]
The is an early version here:

http://mediaplayer.tiddlyspot.com/

**Reply by arunnbabu81** [reply]
> The is an early version here:
> 
> http://mediaplayer.tiddlyspot.com/

This looks interesting. Will get back to you after checking in detail.

**Reply by arunnbabu81** [reply]
Which all tiddlers from the mediaplayer wiki should I use to get mediaplayer working in my wiki

**Reply by arunnbabu81** [reply]
I think it doesn't work with the new versions of TW. I installed the mplayer, mosaic, small tools, taglist plug ins from the mediaplayer wiki.

But the playlist tiddler shows like this

![image](https://user-images.githubusercontent.com/67494083/151594406-51f31598-4d4e-4ace-a2c1-9547c3afd8f9.png)

**Reply by buggyj** [reply]
I will see if I can get to working.

**Reply by buggyj** [reply]
I have uploaded a new version that works with 5.2.0

**Reply by arunnbabu81** [reply]
> I have uploaded a new version that works with 5.2.0

Its working. Thank you for the speedy release. I will get back to you after testing it in detail

**Reply by arunnbabu81** [reply]
> I have uploaded a new version that works with 5.2.0

I have created a [new issue](https://github.com/buggyj/mediaplayer/issues/1#issue-1123358332) in the mediaplayer github page

**Comment by buggyj** [comment]
Like this
![image](https://user-images.githubusercontent.com/2855795/151204773-4246dfae-c707-4f4a-adf3-c618bd825dc5.png)



# The contains are: (that is the letter v)


`#rule=/.*v=(.*?)$/g`
`#term=$1`
`$yt-id=((*@pageRef/#rule/#term*))`

**Reply by buggyj** [reply]
![image](https://user-images.githubusercontent.com/2855795/151205327-193f24a9-96fe-45a9-96ea-62cf09259e04.png)

**Reply by arunnbabu81** [reply]
Awesome. For most of my queries there were solutions in tiddlyclip. I wish more people adopt tiddlyclip into their wiki. It makes our wiki life so easy.
